lib/Service/QueueService.php

```diff
@@ -47,6 +48,14 @@ public function insertIntoQueue(string $model, QueueFile $file) : void {
 		if ($this->config->getAppValue('recognize', $model.'.enabled', 'false') !== 'true') {
 			return;
 		}
+		try {
+			// Only ad to queue if it's not in there already
+			if ($this->queueMapper->existsQueueItem($model, $file) !== null) {
+				return;
+			}
+		} catch (MultipleObjectsReturnedException $e) {
+			return;
+		}
```

Comment:
With above changes this could then turn into:
```suggestion
		// Only ad to queue if it's not in there already
		if ($this->queueMapper->existsQueueItem($model, $file)) {
			return;
		}
```